src/check/constrain/constraint/builder.rs

```diff
@@ -60,6 +61,15 @@ impl ConstrBuilder {
         trace!("Inserted {var} in constraint builder: {var} => {mapped_var}");
     }
 
+    /// Get a name for a temporary type.
+    ///
+    /// Useful for when we don't know what a type should be during the generation stage.
+    /// The unification stage should then identify these.
+    pub fn temp_var(&mut self) -> String {
```

Comment:
Rename to `temp_ty` perhaps.
In fact, if we always convert the `String` to a `Name`, we can instead return a `Name`.